src/main/java/zd/zero/waifu/motivator/plugin/assets/RemoteAssetDefinitionService.kt

```diff
@@ -11,17 +11,44 @@ abstract class RemoteAssetDefinitionService<T : AssetDefinition, U : Asset>(
 ) {
     private val random = Random(System.currentTimeMillis())
 
+    fun getAssetByGroupId(
+        groupId: UUID,
+        category: WaifuAssetCategory
+    ): Optional<U> =
+            remoteAssetManager.supplyLocalAssetDefinitions()
+                .find {
+                    it.groupId == groupId
+                }.toOptional()
+                .map {
+                    resolveAsset(category, it)
+                }.orElseGet {
+                    remoteAssetManager.supplyRemoteAssetDefinitions()
```

Comment:
I figured it would be useful to download the related asset rather than falling back to a random asset. This way the grouped asset gets downloaded and we don't have to wait for it to get downloaded on the random pick.